blog/2023-04-24-naming-translation-keys.md

```diff
@@ -0,0 +1,171 @@
+| -------- | ------------------------------ |
+| Cancel   | `common.form.button.cancel`    |
+| Credits  | `common.label.credits`         |
+| Standard | `common.pricing_plan.standard` |
+
+The translation keys above are used across multiple pages or sections of the application, so it makes sense for them to include the `common` prefix in the key name.
+
+For instance, `common.form.button.cancel` is likely to be used in **various forms across different pages** of an application, making it a common term.
+
+Similarly, `common.label.credits` could refer to the **amount of credits a user has in their account** and may appear in various application sections.
+
+Finally, `common.pricing_plan.standard` could be a frequently used term, as **it denotes the pricing plan "Standard"** that may be mentioned across different pages.
+
+### Use Nested Structure
+
+Organizing translation keys into **nested structures is an effective way to group similar strings**. This helps maintain consistency in the naming convention, especially in larger projects with many translation keys.
+
+A nested structure is a structure that separates keys into nested levels, **each level representing a different aspect of the application**. For example, the first level could be the page or section of the application, and the second level could be the type of string (e.g., button text, title, subtitle, etc.).
+
+:::tip
+
+🐁 Tolgee supports nested structures for translation keys. You can use the `.` character to separate the levels of the nested structure, for example, `footer.contact_info.phone_number`. When [exporting the translations](/platform/projects_and_organizations/export), Tolgee will automatically convert the keys into a nested structure, as shown below.
+
+:::
+
+**Example**:
+
+```json
+{
+  "header": {
+    "title_text": "Welcome to our Website!",
+    "logo_alt_text": "Our company logo"
+  },
+  "error": {
+    "field_required": "This field is required.",
+    "incorrect_url_format": "Please enter a valid URL."
+  },
+  "footer": {
+    "copyright_text": "© 2023 Our Company. All rights reserved.",
+    "contact_info": {
+      "phone_number": "555-123-4567",
+      "email_address": "info@ourcompany.com"
+    }
+  }
+}
+```
+
+## Conclusion
+
+In conclusion, accurately naming translation keys is crucial for efficient management of the translation process across languages and updates to the application. Appropriately named translation keys **provide context, clarity, and consistency**, which help **avoid potential misunderstandings and errors** that could result from vague or poorly named keys.
+
+When naming translation keys, it's critical to use a **clear and consistent naming convention, meaningful and unambiguous names**, and to keep the naming simple. By following best practices for naming translation keys, developers and localization managers can ensure translations are accurate and consistent, **saving time and effort for both translators and developers**.
```

Comment:
Would be great to add some CTA at the end of every article, to boost an engagement, or offer help to create a long-term relationship. 

Here is an example from some of those: 💡 Do you have any suggestions on how to make the localization process with Tolgee better? [Let us know](mailto:info@tolgee.io) or create a pull request on [GitHub](https://github.com/tolgee/tolgee-platform).

In case of this article, you can e.g. invite the readers to join our Slack community (link: Tolg.ee/slack). If they have any question regarding the best practices of naming of translation keys, or a specific question related to their project, they can discuss it there :)